importer/helper.go

```diff
@@ -259,21 +260,40 @@ func updateAccounting(db idb.IndexerDb, frozenCache map[uint64]bool, filter idb.
 	rounds = 0
 	txnCount = 0
 	lastlog := time.Now()
+	roundStart := lastlog
 	act := accounting.New(frozenCache)
 	txns := db.YieldTxns(context.Background(), filter.StartRound)
 	currentRound := uint64(0)
 	roundsSeen := 0
 	lastRoundsSeen := roundsSeen
 	txnForRound := 0
 	var blockHeaderPtr *types.BlockHeader = nil
-	for txn := range txns {
-		maybeFail(txn.Error, l, "updateAccounting txn fetch, %v", txn.Error)
-		if txn.Round != currentRound {
-			if blockHeaderPtr != nil && txnForRound > 0 {
+
+	commit := func() {
+		if blockHeaderPtr != nil {
+			// Don't commit if there were no transactions.
+			if txnForRound > 0 {
 				err := db.CommitRoundAccounting(act.RoundUpdates, currentRound, blockHeaderPtr)
 				maybeFail(err, l, "failed to commit round accounting")
 			}
 
+			// record metric
+			now := time.Now()
+			dt := now.Sub(roundStart)
+			roundStart = now
+			metrics.BlockImportTimeHistogramSeconds.Observe(dt.Seconds())
```

Comment:
Previously we measured import + accounting time, now we measure only accounting time. What is the reason for the change?

Reply:
Good point, maybe I should move it back.

My thinking was that it's possible for this function to process multiple rounds. I suppose for the tests I'm running (and healthy deployments) that wouldn't happen.

Reply:
Yes, in a normal scenario we import one block and account one block.